src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -369,34 +510,172 @@ void QgsRelationEditorWidget::toggleEditing( bool state )
 
 void QgsRelationEditorWidget::updateUi()
 {
-  if ( !mRelation.isValid() || !mFeature.isValid() )
+  if ( !mRelation.isValid() || mFeatureList.isEmpty() || !mFeatureList.first().isValid() )
     return;
 
   if ( !isVisible() )
     return;
 
-  const QgsFeatureRequest request = mRelation.getRelatedFeaturesRequest( mFeature );
+  mFormViewButton->setVisible( !multiEditModeActive() );
+  mTableViewButton->setVisible( !multiEditModeActive() );
+  mMultiEditInfoLabel->setVisible( multiEditModeActive() );
 
-  if ( mNmRelation.isValid() )
+  if ( !multiEditModeActive() )
   {
-    QgsFeatureIterator it = mRelation.referencingLayer()->getFeatures( request );
-    QgsFeature fet;
-    QStringList filters;
+    mStackedWidget->setCurrentWidget( mDualView );
 
-    while ( it.nextFeature( fet ) )
+    const QgsFeatureRequest request = mRelation.getRelatedFeaturesRequest( mFeatureList.first() );
+
+    if ( mNmRelation.isValid() )
     {
-      QString filter = mNmRelation.getReferencedFeatureRequest( fet ).filterExpression()->expression();
-      filters << filter.prepend( '(' ).append( ')' );
-    }
+      QgsFeatureIterator it = mRelation.referencingLayer()->getFeatures( request );
+      QgsFeature fet;
+      QStringList filters;
 
-    QgsFeatureRequest nmRequest;
-    nmRequest.setFilterExpression( filters.join( QLatin1String( " OR " ) ) );
+      while ( it.nextFeature( fet ) )
+      {
+        QString filter = mNmRelation.getReferencedFeatureRequest( fet ).filterExpression()->expression();
+        filters << filter.prepend( '(' ).append( ')' );
+      }
 
-    initDualView( mNmRelation.referencedLayer(), nmRequest );
+      QgsFeatureRequest nmRequest;
+      nmRequest.setFilterExpression( filters.join( QLatin1String( " OR " ) ) );
+
+      initDualView( mNmRelation.referencedLayer(), nmRequest );
+    }
+    else if ( mRelation.referencingLayer() )
+    {
+      initDualView( mRelation.referencingLayer(), request );
+    }
   }
-  else if ( mRelation.referencingLayer() )
+  else
```

Comment:
Could you move the updateUi code specific to multieditmode to its own method, in order to keep the code in updateUi short and simple ?

Reply:
Yes I'll move it